settings/resource_settings_test.go

```diff
@@ -0,0 +1,215 @@
+package settings
+
+import (
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/qa"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestQueryCreate(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/settings/types/default_namespace_ws/names/default?",
+				Status:   404,
+				Response: apierr.APIErrorBody{
+					ErrorCode: "NOT_FOUND",
+					Message:   "SomeMessage",
+					Details: []apierr.ErrorDetail{{
+						Type: "type.googleapis.com/google.rpc.ErrorInfo",
+						Metadata: map[string]string{
+							"etag": "etag1",
+						},
+					}},
+				},
+			},
+			{
+				Method:   "PATCH",
+				Resource: "/api/2.0/settings/types/default_namespace_ws/names/default",
+				Status:   200,
+				ExpectedRequest: settings.UpdateDefaultWorkspaceNamespaceRequest{
+					AllowMissing: true,
+					FieldMask:    "namespace.value",
+					Setting: &settings.DefaultNamespaceSetting{
+						Etag: "etag1",
+						Namespace: settings.StringMessage{
+							Value: "namespace_value",
+						},
+						SettingName: "default",
+					},
+				},
+				Response: &settings.DefaultNamespaceSetting{
+					Etag: "etag2",
+					Namespace: settings.StringMessage{
+						Value: "namespace_value",
+					},
+					SettingName: "default",
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/settings/types/default_namespace_ws/names/default?etag=etag2",
+				Status:   200,
+				Response: &settings.DefaultNamespaceSetting{
+					Etag: "etag3",
```

Comment:
This GET would still return etag2, since the PATCH above returns etag2. (GET does not create a new version)

Reply:
Updated